### PR TITLE
Verify that new email is unique when editing a subscriber MAILPOET-4259

### PR DESCRIPTION
**Pull request by brezocordero**
With this PR, it throws an exception and displays a friendlier message if the new email already exists.

To test:
- Go to MailPoet>Subscribers>Edit a subscriber
- Enter an email that already exists and click save
- Check you see the message: `A subscriber with E-mail '{$email}' already exists.` instead of the DB exception

When working on this issue I found a tiny bug and fixed it:
To test:
- Go to MailPoet>Subscribers>Import>Proceed>Paste data
- Enter data for 1 subscriber `mary@jane.com, mary, jane` (doesn't really matter) click next.
- Click on 'Create a new list', enter the name of a list that already exists.
- Check that instead of seeing `An unknown error has occurred` you see `Another record already exists. Please specify a different "name".`

[MAILPOET-4259]

[MAILPOET-4259]: https://mailpoet.atlassian.net/browse/MAILPOET-4259?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by johnolek — CHANGES_REQUESTED** [review]
Looks good overall! The only change I would definitely like to see is to ensure that we're still trying to log errors if an unknown exception occurs during a save (more details in my comment).

**Comment by johnolek** [comment]
@brezocordero I'm not sure I'm understanding how to reproduce this issue you describe:

> When working on this issue I found a tiny bug and fixed it:
> To test:
> 
> Go to MailPoet>Subscribers>Import>Proceed>Paste data
> Enter data for 1 subscriber mary@jane.com, mary, jane (doesn't really matter) click next.
> Check that instead of seeing An unknown error has occurred you see Another record already exists. Please specify a different "name"

When I paste an email for an existing subscriber and click next, I don't see an error at all. Then on the import step, if I choose `No` for `Update existing subscribers’ information, like first name, last name, etc` and click `Import`, it just tells me that `No subscribers were added or updated.` Is there something I could be missing?

**Comment by brezocordero** [comment]
> I'm not sure I'm understanding how to reproduce this issue you describe:

So sorry @johnolek, I forgot a step in the test! I have updated the issue.

File: mailpoet/lib/API/JSON/v1/Subscribers.php [hunk 1]
```diff
@@ -146,13 +149,24 @@ public function subscribe($data = []) {
...
     } catch (ValidationException $validationException) {
       return $this->badRequest([$this->getErrorMessage($validationException)]);
+    } catch (ConflictException $conflictException) {
+      return $this->badRequest([
+        APIError::BAD_REQUEST => $conflictException->getMessage(),
+      ]);
+    } catch (\Exception $unknownException) {
+      return $this->badRequest([
+        APIError::UNKNOWN => __('Saving subscriber failed.', 'mailpoet'),
+      ]);
```
[the remark this answers is not in the document]
Comment (brezocordero) [reply mailpoet/lib/API/JSON/v1/Subscribers.php]: Removed the last catch block so that the message gets logged.

**Review by johnolek — APPROVED** [review]
Thanks for making those changes and updating the testing instructions -- this looks good to me!

**Review by veljkho — APPROVED** [review]

**Comment by rohankacharegs** [comment]
GlobalStep Verified

**Pull request merged**